web/secondary_toolbar.js

```diff
@@ -29,173 +29,232 @@
+      var elements = [
+        // Button to toggle the visibility of the secondary toolbar.
+        { element: this.toggleButton, handler: this.toggle },
+        // All items within the secondary toolbar.
+        { element: this.presentationModeButton,
+          handler: this.presentationModeButtonClick },
+        { element: this.openFileButton, handler: this.openFileButtonClick },
+        { element: this.printButton, handler: this.printButtonClick },
+        { element: this.downloadButton, handler: this.downloadButtonClick },
+        { element: this.viewBookmarkButton,
+          handler: this.viewBookmarkButtonClick },
+        { element: this.firstPageButton, handler: this.firstPageButtonClick },
+        { element: this.lastPageButton, handler: this.lastPageButtonClick },
+        { element: this.pageRotateCwButton,
+          handler: this.pageRotateCwButtonClick },
+        { element: this.pageRotateCcwButton,
+          handler: this.pageRotateCcwButtonClick },
+        { element: this.toggleHandToolButton,
+          handler: this.toggleHandToolButtonClick },
+        { element: this.documentPropertiesButton,
+          handler: this.documentPropertiesButtonClick }
+      ];
+
+      for (var item in elements) {
+        var element = elements[item].element;
+        if (element) {
+          element.addEventListener('click', elements[item].handler.bind(this));
+        }
       }
-    }
+    },
 
-    // Tracking hand tool menu item changes.
-    var isHandToolActive = false;
-    this.eventBus.on('handtoolchanged', function (e) {
-      if (isHandToolActive === e.isActive) {
+    bindHandToolListener: function SecondaryToolbar_bindHandToolListener() {
+      var isHandToolActive = false;
+      this.eventBus.on('handtoolchanged', function (e) {
+        if (isHandToolActive === e.isActive) {
+          return;
+        }
+        isHandToolActive = e.isActive;
+        if (isHandToolActive) {
+          this.toggleHandToolButton.title =
+            mozL10n.get('hand_tool_disable.title', null, 'Disable hand tool');
+          this.toggleHandToolButton.firstElementChild.textContent =
+            mozL10n.get('hand_tool_disable_label', null, 'Disable hand tool');
+        } else {
+          this.toggleHandToolButton.title =
+            mozL10n.get('hand_tool_enable.title', null, 'Enable hand tool');
+          this.toggleHandToolButton.firstElementChild.textContent =
+            mozL10n.get('hand_tool_enable_label', null, 'Enable hand tool');
+        }
+      }.bind(this));
+    },
+
+    open: function SecondaryToolbar_open() {
+      if (this.opened) {
+        return;
+      }
+      this.opened = true;
+      this.toggleButton.classList.add('toggled');
+      this.toolbar.classList.remove('hidden');
+    },
+
+    close: function SecondaryToolbar_close(target) {
+      if (!this.opened) {
+        return;
+      } else if (target && !this.toolbar.contains(target)) {
```

Comment:
It appears that there are no longer any callsite that passes an argument to `SecondaryToolbar_close`. Hence I think that the `else if`, and as well as the parameter, can now be removed.